swift/extractor/visitors/PatternVisitor.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "swift/extractor/SwiftDispatcher.h"
+#include <swift/AST/ASTVisitor.h>
+
+namespace codeql {
+
+// swift code lacks default implementations of visitor for some entities. We can add those here
+// while we do not have yet all implemented. This is copy and pasted from the corresponding Expr
+// code in swift/AST/ASTVisitor.h
+template <typename ImplClass, typename PatternRetTy = void, typename... Args>
```

Comment:
Nit: I'd rename `ImplClass` to something like `CrtpSubclass` to make it more obvious it's the [Curiously recurring template pattern](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern) rather than e.g. PImpl.

Reply:
It's merely a copy-paste from Swift's source code, but I'll update it.

Reply:
Done.